libs/langgraph/langgraph/func/__init__.py

```diff
@@ -174,6 +178,84 @@ async def agen_wrapper(
             checkpointer=checkpointer,
             store=store,
             config_type=config_schema,
+            graph=_entrypoint_graph(bound),
         )
 
     return _imp
+
+
+def _find_children(
+    candidate: Runnable, parent: Node
+) -> Iterator[tuple[Node, Union[Callable, PregelProtocol]]]:
+    from langchain_core.runnables.utils import get_function_nonlocals
+
+    from langgraph.utils.runnable import (
+        RunnableCallable,
+        RunnableLambda,
+        RunnableSeq,
+        RunnableSequence,
+    )
+
+    candidates: list[Runnable] = [candidate]
+
+    for c in candidates:
+        print(c, type(c))
+        if callable(c) and getattr(c, "_is_pregel_task", False) is True:
+            yield (parent, c)
+        elif isinstance(c, PregelProtocol):
+            yield (parent, c)
+        elif isinstance(c, RunnableSequence) or isinstance(c, RunnableSeq):
+            candidates.extend(c.steps)
+        elif isinstance(c, RunnableLambda):
+            candidates.extend(c.deps)
+        elif isinstance(c, RunnableCallable):
+            if c.func is not None:
+                candidates.extend(
+                    nl.__self__ if hasattr(nl, "__self__") else nl
+                    for nl in get_function_nonlocals(c.func)
+                )
+            elif c.afunc is not None:
+                candidates.extend(
+                    nl.__self__ if hasattr(nl, "__self__") else nl
+                    for nl in get_function_nonlocals(c.afunc)
+                )
+
+
+def _entrypoint_graph(entrypoint: Runnable, xray: int = 0) -> Graph:
```

Comment:
_question_: worth doing this on-demand rather than during initialization?

Reply:
yea probably, will change